### PR TITLE
Use same code for setting up cli/non-cli formatter

### PR DESCRIPTION
**Pull request by thisch**
A method _create_formatter was introduced that is used for both the
log_cli_formatter and the log_formatter.

Consequences of this commit are:
* Captured logs that are output for each failing test are formatted
  using the ColoredLevelFromatter.
* caplog.text now contains ansi escape sequences.
* The formatter used for writing to a file still uses the non-colored
  logging.Formatter class.

- [x] Target the `master` branch for bug fixes, documentation updates and trivial changes.
- [x] Target the `features` branch for new features and removals/deprecations.
- [ ] Include documentation when adding new features.
- [ ] Include new tests or update existing tests when applicable.

Unless your change is trivial or a small documentation fix (e.g.,  a typo or reword of a small section) please:

- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/master/changelog/README.rst) for details.
- [x] Add yourself to `AUTHORS` in alphabetical order;

### CODE REVIEW

**Comment by thisch** [comment]
I've just noticed that the ANSI escape sequences are also part of junit xml files now. I'm not sure if all tools that parse those files support ANSI escape sequences.

**Comment by blueyed** [comment]
> I'm not sure if all tools that parse those files support ANSI escape sequences.

I do not think they should (have to) care about this.

**Comment by thisch** [comment]
> > As a consequence caplog.text contains the ANSI
> > escape sequences used for coloring the level names now.
> 
> I definitely don't think this is a good idea; users who use `caplog.text` definitely don't expect those ANSI escape sequences related to color there. How about creating a new attribute, say, `colored_text`?

SInce the log level colors are added internally, they should IMO not be exposed to the user via a caplog API. Therefore, I'll remove the ansi escape sequences from the text property, but I'll not add a colored_text property.

**Comment by thisch** [comment]
I had a look at the support of ANSI escape sequences in the junit plugin of jenkins: They are unfortunately not yet rendered correctly. I'll therefore remove the ansi escape sequences also from the generated junit files.

**Comment by thisch** [comment]
> > > As a consequence caplog.text contains the ANSI
> > > escape sequences used for coloring the level names now.
> > 
> > 
> > I definitely don't think this is a good idea; users who use `caplog.text` definitely don't expect those ANSI escape sequences related to color there. How about creating a new attribute, say, `colored_text`?
> 
> SInce the log level colors are added internally, they should IMO not be exposed to the user via a caplog API. Therefore, I'll remove the ansi escape sequences from the text property, but I'll not add a colored_text property.

Done.

File: testing/logging/test_reporting.py [hunk 1]
```diff
@@ -1084,3 +1084,48 @@ def test_second():
...
+    """
+    testdir.makepyfile(
+        """
+        import logging
+
+        logger = logging.getLogger(__name__)
+
+        def test_foo(caplog):
+            logger.info('text going to logger from call')
+            assert '\x1b' not in caplog.text
```
Comment (thisch) [inline testing/logging/test_reporting.py]: I made sure that this test fails if my changes to caplog.text are removed.

**Comment by thisch** [comment]
Can we merge this?

**Comment by nicoddemus** [comment]
I will merge this in a couple of hours unless someone steps in to take a second look. 👍

File: changelog/5311.feature.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Captured logs that are output for each failing test are formatted using the
+ColoredLevelFromatter.
```
Comment (blueyed) [inline changelog/5311.feature.rst]:
```suggestion
ColoredLevelFormatter.
```

**Comment by thisch** [comment]
I know that I can merge it, but I just wanted to make sure that you have no objections against merging this.

**Comment by blueyed** [comment]
@thisch 
:+1: 

Thanks!

**Pull request merged**